chore(docs): getting started (#403)
pmndrs/jotai: docs/api/core.md

```

The `updateValue` takes just one argument, which will be passed to the third argument of writeFunction of the atom. The behavior totally depends on how the writeFunction is implemented.

---

# How atom dependency works

> To begin with, let's explain this. In the current implementation, every time we invoke the "read" function, we refresh dependencies. For example, If A depends on B, it means that B is a dependency of A, and A is a dependent of B.

Author: This is to explain "dependencies" and "dependents". Maybe, a diagram would help though.

> To begin with, let's explain this. In the current implementation, every time we invoke the "read" function, we refresh dependencies. For example, If A depends on B, it means that B is a dependency of A, and A is a dependent of B.

Author: ```suggestion
To begin with, let's explain this. In the current implementation, every time we invoke the "read" function, we refresh dependencies and dependents. For example, If A depends on B, it means that A is a dependent of B.
```

```js
const uppercaseAtom = atom((get) => get(textAtom).toUpperCase())
```

The read function is the first parameter of the atom.

> The dependency will initially be empty. On first use, we run the read function and know that uppercaseAtom depends on textAtom. textAtom is the dependency of uppercaseAtom. So, add uppercaseAtom to the dependents of textAtom.

Reviewer: ```suggestion
The dependency will initially be empty. On first use, we run the read function and know that `uppercaseAtom` depends on `textAtom`. `textAtom` has a dependency on `uppercaseAtom`. So, add `uppercaseAtom` to the dependents of `textAtom`.
```

Author: okay "has a dependency on"

When we re-run the read function (because its dependency (=textAtom) is updated),
the dependency is built again, which is the same in this case. We then remove stale dependents and replace with the latest one.

# Atoms can be created on demand

Basic examples in readme only show defining atoms globally outside components.
There is no restrictions about when we create an atom.
As long as we know atoms are identified by their object referential identity,
it's okay to create them at anytime.

If you create atoms in render functions, you would typically want to use

> some hooks like `useRef` or `useMemo`.

Reviewer: ```suggestion
a hook like `useRef` or `useMemo` for memoization. If not, the atom would be created everytime the component renders.
```

You can create an atom and store it wth `useState` or even in another atom.
See an example in [issue #5](https://github.com/pmndrs/jotai/issues/5).

You can cache atoms somewhere globally.
See [this example](https://twitter.com/dai_shi/status/1317653548314718208) or
[that example](https://github.com/pmndrs/jotai/issues/119#issuecomment-706046321).

> Check [`atomFamily`](../api/utils.md#atomfamily) in utils too.

Reviewer: ```suggestion
Check [`atomFamily`](../api/utils.md#atomfamily) in utils for parameterized atoms.
```
